WordPress/gutenberg
# Social Link: Update Reddit icon and color to match brand guidelines

## What?
Updated the Reddit icon and color used in the Social Icons block.

## Why?
The Reddit icon and color currently used in the Social Icons block do not match the [Reddit brand guidelines](https://www.redditinc.com/brand). 

## How?
Update the SVG and color.

## Testing Instructions
1. Insert the Social Icons block and add the Reddit icon. Notice the new logo and updated color. 

## Screenshots or screencast 

Before:
![image](https://user-images.githubusercontent.com/4832319/185142165-14ce737f-b23d-493a-9033-e2aa648a603c.png)

After:
![image](https://user-images.githubusercontent.com/4832319/185142053-79f8b39f-55be-41c2-b4fc-167a0b231de5.png)
